tools/serverpod_cli/lib/src/internal_tools/analyze_pubspecs.dart

```diff
@@ -2,10 +2,65 @@ import 'dart:io';
 
 import 'package:pub_api_client/pub_api_client.dart';
 import 'package:pubspec_parse/pubspec_parse.dart';
+import 'package:serverpod_cli/src/util/pubspec_helpers.dart';
 
 /// The internal tool for analyzing the pubspec.yaml files in the Serverpod
 /// repo.
 Future<void> performAnalyzePubspecs(bool checkLatestVersion) async {
+  if (!_isServerpodRootDirectory()) {
+    print('Must be run from the serverpod repository root');
+    exit(1);
+  }
+
+  var pubspecFiles = findPubspecsFiles(Directory.current,
+      ignorePaths: ['/templates/pubspecs/', '/test_assets/']);
+
+  var dependencies = _getDependencies(pubspecFiles);
+
+  var missmatchedDeps = _findMissmatchedDependencies(dependencies);
+
+  if (missmatchedDeps.isNotEmpty) {
+    _printMissmatchedDependencies(missmatchedDeps, dependencies);
+    exit(1);
+  }
+
+  print('Dependencies match.');
+
+  if (checkLatestVersion) {
+    await _checkLatestVersion(dependencies);
+  }
+}
+
+Future<void> _checkLatestVersion(
+    Map<String, List<_ServerpodDependency>> dependencies) async {
+  print('Checking latest pub versions.');
+  try {
+    var pub = PubClient();
+    for (var depName in dependencies.keys) {
+      var deps = dependencies[depName]!;
+      var depVersion = deps.first.version;
+      var latestPubVersion = _latestStableVersion(
+        await pub.packageVersions(depName),
+      );
+
+      if (depVersion != '^$latestPubVersion') {
+        print(depName);
+        print('  local: $depVersion');
+        print('  pub:   ^$latestPubVersion');
+        print('  found in:');
+        for (var dep in deps) {
+          print('   - ${dep.serverpodPackage}');
+        }
+      }
+    }
+    pub.close();
+  } catch (e) {
+    print('Version check failed.');
+    print(e);
+  }
+}
+
+bool _isServerpodRootDirectory() {
```

Comment:
Maybe this should be public and placed in the util directory? Will most likely be usable from many places in our code base. Like: 
```dart
bool isServerpodRootDirectory([directory = Directory.current()]) {
  ...
}```